### ISSUE TITLE
False positive circular ref on large JSON payloads

### ISSUE THREAD

**Issue by bitomaxsp** [pb33f/libopenapi#256]
Hi Dave.

We have a case with unusually large payload where `0.15.12` (0.13.3 work fine ) detects circular reference of weird kind.

Error message:
```sh
infinite circular reference detected: PrisModuleDimensionCheckedDimensionCheckMeasurements: endpoint.yaml -> PrisModuleDimensionCheckedRequest -> PrisModuleDimensionCheckedBody -> PrisModuleDimensionCheckedTrace -> PrisModuleDimensionCheckedOperations -> PrisModuleDimensionCheckedDimensionCheck -> PrisModuleDimensionCheckedDimensionCheckMeasurements -> Float64 [0:0]
```

[lib2.zip](https://github.com/pb33f/libopenapi/files/14358329/lib2.zip)

Attached is minimal repo sample. You can run the test and see the error. I tried to debug, but i couldn't figure out why it is happening. Would really appreciate you help :)

**Comment by daveshanley** [comment]
`0.15.12` adds some code to correct detecting circular references, before it was missing a number of them because it was skipping over remaining journeys after it found a path with a reference.

It was reported here: https://github.com/daveshanley/vacuum/issues/459

Thats why you're seeing a change, however you could have discovered another bug that has now been exposed by the fixing of another.

I'll open load up the example and take a look for you.

**Comment by daveshanley** [comment]
This is interesting, your set up is hitting a safety net that prevents run-away circles from being detected, there is a hard limit of 500 levels. Your models are hitting that number, except I only see seven levels of depth, which means there is a definite bug in here, and I think I know where it is. 

Stay tuned for more updates.

**Comment by daveshanley** [comment]
The problem was simple, there is a depth counter that was not being decremented, your complex models where fooling the resolver into thinking it was hundreds of levels deep, when really, it had just been processing hundreds of sibling nodes. The depth counter was not being decremented.

**Comment by daveshanley** [comment]
Fixed in `v0.15.13`

**Comment by bitomaxsp** [comment]
Confirming it is fixed! thanks a lot
